src/flood.rs

```diff
@@ -0,0 +1,78 @@
+use crate::view::layer::LayerCoords;
+use crate::view::{Snapshot, View, ViewResource};
+use rgx::color::{Rgba8};
+use rgx::kit::shape2d::{Fill, Rotation, Shape, Stroke};
+use rgx::kit::ZDepth;
+use rgx::rect::Rect;
+
+pub fn flood_fill(
+    view: &View<ViewResource>,
+    starting_point: LayerCoords<u32>,
+    replacement_color: Rgba8,
+) -> Option<Vec<Shape>> {
+    let (snapshot, pixels) = view.current_snapshot(view.active_layer_id)?;
+    let mut canvas = pixels.clone().into_rgba8();
+    let bounds = snapshot.extent.rect();
+    let target_color = get_color(snapshot, &mut canvas, starting_point)?.clone();
+
+    if target_color == replacement_color {
+        return None;
+    }
+
+    let mut points = Vec::new();
+    let mut queue = vec![starting_point];
+
+    while let Some(point) = queue.pop() {
+        if let Some(c) = get_color(snapshot, &mut canvas, point) {
+            if *c == target_color {
+                *c = replacement_color;
```

Comment:
Looking at the code, I'm not sure this is used? If that's the case, you can remove the mutability requirement on `canvas``.

Reply:
Yeah, it's a little sneaky, but the `canvas` vec is being used to keep track of which pixels we've already filled. Setting `c` here is what keeps us from trying to fill the same pixels when `neighbors` spits them out again.

Reply:
Ah, I missed that. Cool.